dbconnect.py

```diff
@@ -0,0 +1,72 @@
+#TODO Needs more work to make it complete
+
+import psycopg2
+
+
+class DBConnect(object):
+    """
+    Class for connecting to a postgresql database using a with statement
+    """
+
+    def __init__(self, dbhost='', db='', dbuser='', dbpass='', dbport=3306, autocommit=False,
+                 *args, **kwargs):
+        try:
+            self.conn = psycopg2.connect(host=dbhost, database=db, user=dbuser,
+                                         password=dbpass, port=dbport)
+            self.cursor = self.conn.cursor()
+        except psycopg2.Error:
+            raise
+
+        self.autocommit = autocommit
+        self.fetcharr = []
+
+    def execute(self, sql_str):
```

Comment:
accepting a straight SQL string for everything would open up the possibility of SQL injection attacks.  queries that accept parameters should always be parametarized through the cursor object... FYI.  Not saying that would become an issue here given where this class is located and how it's used, just making you aware.

http://stackoverflow.com/questions/1466741/parameterized-queries-with-psycopg2-python-db-api-and-postgresql
-